### ISSUE TITLE
Unbale to find ID of undefined

### ISSUE THREAD

**Issue by nikhilgrag** [rangle/angular-devtools#774]
Angular version 10.2.4

When I try to inspect any component with Angular Devtools on my angular app, I get the following error.
Also It doesn't show any component in the component list under the Angular devtools tab in chrome.

`backend.js:1 Unable to find ID of undefined
 getDirectiveId @ backend.js:1
 elementMouseOver @ backend.js:1
 invokeTask @ polyfills.js:412
 runTask @ polyfills.js:180
 invokeTask @ polyfills.js:493
 invokeTask @ polyfills.js:1634
 globalZoneAwareCaptureCallback @ polyfills.js:1692
`

**Comment by sumitarora** [comment]
@nikhilgrag Is IVY enabled on the application? Also Is it possible if you can provide a demo application replicating the issue?

**Comment by TheCoderFish** [comment]
I am facing the same issue, I will try it on a blank app and see if the issue persists, for now, see if this screenshot of the stack trace is of any help
![Screenshot 2021-05-19 at 9 16 06 AM](https://user-images.githubusercontent.com/18303446/118753727-31c2ab00-b883-11eb-886a-6538dce42c34.png)

**Comment by vmalvaro** [comment]
Facing the same issue, with Angular 11 in dev and ivy enabled.

**Comment by nikhilgrag** [comment]
@sumitarora yes I have Ivy enabled.

While debugging I found out the issue for why it is not working for me. Not sure if everyone has the same issue.

So, AngularDev tools don't work for

- Overlay components
- components where 'ViewContainerRef' is injected in the constructor.
`constructor(
        private vcr: ViewContainerRef,
    )`

After removing ViewContainerRef from my app component constructor, devtools started working.

**Comment by vmalvaro** [comment]
Can confirm that after removing `private readonly vcr: ViewContainerRef` dev tools works again.

**Comment by mgechev** [comment]
This seems like a known issue with v11 and older versions. Would you try updating to v12 and run DevTools again? We introduced new debugging APIs that help us find the render tree more accurately.

**Comment by mgechev** [comment]
FYI, I'm labeling as a P3 because it should be working properly with v12. Let me know if that's not the case.

**Comment by nikhilgrag** [comment]
@mgechev I checked with v12. It has the same error with injecting `private vcr: ViewContainerRef` in the constructor.

**Comment by nikhilgrag** [comment]
@mgechev You are right, it works perfectly with v12, even when ViewContainerRef is injected in the component's constructor.

**Comment by mgechev** [comment]
@vmalvaro would you confirm it works with v12?

**Comment by vmalvaro** [comment]
@mgechev Can't confirm, as we are not ready to move our project to v12 yet.
I'll let you know what happens, once we move to Angular 12 in the next few weeks.

**Comment by jamisonbryant** [comment]
Is there any workaround for projects that cannot upgrade to Angular 12 and cannot remove `ViewContainerRef` from their constructors? I'm working as part of a team and do not have decision making power over either of those aspects.
